src/System.Text.Formatting/System/Text/Parsing/InvariantParser_int.cs

```diff
@@ -0,0 +1,135 @@
+            bool negative = false;
+
+            if (utf8Text[index] == '-')
+            {
+                negative = true;
+                index++;
+                bytesConsumed++;
+            }
+
+            for (int byteIndex = index; byteIndex < utf8Text.Length; byteIndex++) // loop through the byte array
+            {
+                if (utf8Text[byteIndex] < '0' || utf8Text[byteIndex] > '9') // if the next character is not a digit
+                {
+                    if (bytesConsumed == 0) // check to see if we've processed any digits at all
+                    {
+                        value = default(int); // if we haven't, set value to 0 and return false
+                        return false;
+                    }
+                    else
+                    {
+                        if (negative) // We check if the value is negative at the very end to save on comp time
+                        {
+                            value *= -1;
+                        }
+                        return true; // otherwise return true
+                    }
+                }
+                int candidate = (value * 10); // left shift the value
+                candidate += (utf8Text[byteIndex] - '0'); // parse the current digit to an int and add it to the temporary value
+                if (candidate >= value) // if it was a digit 0-9, this should be true
+                {
+                    value = candidate;
+                }
+                else // for signed types this will occur at the min values as overflow occurs during addition, so we handle that
+                {
+                    if (candidate == int.MinValue)
+                    {
+                        bytesConsumed++;
+                        value = candidate;
+                        return true;
+                    }
+                    if (negative) // We check if the value is negative at the very end to save on comp time
+                    {
+                        value *= -1;
+                    }
+                    return true;
+                }
+                bytesConsumed++; // increment the number of bytes consumed, then loop
+            }
+
+            if (negative) // We check if the value is negative at the very end to save on comp time
+            {
+                value *= -1;
+            }
+            return true;
+        }
+
+        unsafe public static bool TryParse(byte* utf8Text, int index, out int value, out int bytesConsumed)
+        {
+            value = 0;
+            bytesConsumed = 0;
+            bool negative = false;
+
+            if (utf8Text[index] == '-')
+            {
+                negative = true;
+                index++;
+                bytesConsumed++;
+            }
+
+            for (int byteIndex = index; ; byteIndex++)
+            {
+                if (utf8Text[byteIndex] < '0' || utf8Text[byteIndex] > '9')
+                {
+                    if (bytesConsumed == 0)
```

Comment:
But if you read '-' sign before, bytesConsumed won't be zero, and so it seems like "-A" will parse as zero, but it should fail to parse.